src/toxcore/common_parsers.rs

```diff
@@ -0,0 +1,97 @@
+/*
+    Copyright © 2017 Zetok Zalbavar <zexavexxe@gmail.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+//! Generic nom parsers.
+
+use std::mem::transmute;
+use nom::{IResult, Needed};
+use nom::IResult::*;
+
+/// Recognizes an unsigned 1 byte integer (equivalent to take!(1)
+#[inline]
+pub fn ne_u8(i: &[u8]) -> IResult<&[u8], u8> {
+    if i.len() < 1 {
+        Incomplete(Needed::Size(1))
+    } else {
+        Done(&i[1..], i[0])
+    }
+}
+
+/// Recognizes native endian unsigned 2 bytes integer
+#[inline]
+pub fn ne_u16(i: &[u8]) -> IResult<&[u8], u16> {
+    if i.len() < 2 {
+        Incomplete(Needed::Size(2))
+    } else {
+        let res = unsafe {
+            transmute([i[0], i[1]])
+        };
+        Done(&i[2..], res)
+    }
+}
+
+/// Recognizes native endian unsigned 4 bytes integer
+#[inline]
+pub fn ne_u32(i: &[u8]) -> IResult<&[u8], u32> {
+    if i.len() < 4 {
+        Incomplete(Needed::Size(4))
+    } else {
+        let res = unsafe {
+            transmute([i[0], i[1], i[2], i[3]])
```

Comment:
see examples from [Rust doc](https://doc.rust-lang.org/1.14.0/book/casting-between-types.html#transmute). You may make the code simpler by using `mem::transmute::<[u8; 4], u32>(a)`